client/src/main.jsx

```diff
@@ -1,10 +1,16 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
 import App from "./App.jsx";
-import "./index.css";
+import {store} from './app/store.jsx'
+import { Provider } from "react-redux";
+import ErrorBoundary from "./ErrorBoundary.jsx";
```

Comment:
you are not using error boundery package. please import it and use it - and inside it use the component you created. see docs - https://github.com/bvaughn/react-error-boundary